src/log/log.c

```diff
@@ -2121,6 +2121,19 @@ __wt_log_scan(WT_SESSION_IMPL *session, WT_LSN *lsnp, uint32_t flags,
         WT_SET_LSN(&end_lsn, lastlog, 0);
         WT_ERR(__wt_fs_directory_list_free(session, &logfiles, logcount));
     }
+
+    if (start_range_lsnp != NULL) {
+        /*
+         * Check to see if the user set offset is in alignment, if not correct to the next aligned
+         * log entry.
+         */
+        if (start_range_lsnp->l.offset % allocsize != 0)
+            WT_SET_LSN(&start_lsn, start_range_lsnp->l.file,
+              start_range_lsnp->l.offset + (allocsize - start_range_lsnp->l.offset % allocsize));
+        else
+            WT_ASSIGN_LSN(&start_lsn, start_range_lsnp);
+    }
```

Comment:
This code should no longer be needed at all because it is already handled in the `if (lsnp !=NULL)` line below this one.

Reply:
I added a new start lsnp because I thought it might be useful to align the start offset if the user doesn't give an exact correct offset. I also didn't want to change the underlying functionality of lsnp because I assumed it would be used/test elsewhere with different expected results. Do you think it would be better to just handle the start offset the same way as lsnp is currently handled? That is to just return WT_NOTFOUND if the user doesn't given a correct offset? Or do you think it would be useful to add the align adjustment in handling lsnp?

Reply:
I do not think that aligning to the allocsize is going to be useful. (Also there is a `wt_rduppo2` function for faster rounding up to a power of 2.) The reason is that there is no knowledge of where the next record boundary is other than the previous record's length. So if someone gives an LSN that is in the middle of a record we have no idea where the next or previous record boundary is. We know that a record must start on an `allocsize` boundary (typically 128 bytes) but that is all. The individual record itself could be many megabytes in size.

The caller either sends in a valid LSN or they don't.

Reply:
That said, I do think it would be useful to support N,0 to indicate starting at the beginning of log file N. That is the only special case we should support. Only the starting LSN needs to care about that.

Reply:
Okay yeah, that makes sense, I'll remove the start_range_lsnp and pass the start into the existing lsnp and add that special starting case in there.